### PR TITLE
Merge and edit intro to shader pages

### LINKED ISSUE #4156: Merge intro to shaders pages
Shading reference / Shaders and Your First Shader / What are Shaders? are both intros to shaders. They could use a merge.

### PR DESCRIPTION
**Pull request by NathanLovato**
- Merged Shading reference / Shaders and Your First Shader / What are Shaders? into one intro page
- Edited to better fit the writing guidelines and shorten the text
- Removed tech info about how GPU cores work compared to CPU cores
- Filled to 80 columns

Closes #4156

### CODE REVIEW

File: tutorials/shaders/introduction_to_shaders.rst [hunk 1]
```diff
@@ -3,119 +3,162 @@
...
+Each shader type has different render modes. See the reference for each shader
+type for a complete list of render modes.
+
+Processor functions
+-------------------
+
+Depending on the shader type, you can override different processor functions.
+For ``spatial`` and ``canvas_item``, you have access to ``vertex()``,
+``fragment()``, and ``light()``. For ``particles``, you only have access to
+``vertex()``.
```
Comment (clayjohn) [inline tutorials/shaders/introduction_to_shaders.rst:115]:
```suggestion
``vertex()``. For ''sky'', you only have access to ''fragment()''.
```

File: tutorials/shaders/introduction_to_shaders.rst [hunk 2]
```diff
@@ -3,119 +3,162 @@
...
-threads per wavefront, a wavefront will run on a 10×10 block of pixels together. It will continue to
-run for all pixels in that wavefront until they are complete. Accordingly, if you have one pixel slower
-than the rest (due to excessive branching), the entire block will be slowed down, resulting in massively
-slower render times.
-
-This is different from CPU-based operations. On a CPU, if you can speed up even one
-pixel, the entire rendering time will decrease. On a GPU, you have to speed up the entire wavefront
-to speed up rendering.
+The ``fragment()`` processing function is used to set up the Godot material
+
```
Comment (clayjohn) [inline tutorials/shaders/introduction_to_shaders.rst:137]: extra white space here

File: tutorials/shaders/introduction_to_shaders.rst [hunk 3]
```diff
@@ -3,119 +3,162 @@
...
+parameters per pixel. This code runs on every visible pixel the object or
+primitive draws. It is only available in ``spatial`` and ``canvas_item`` shaders.
```
Comment (clayjohn) [inline tutorials/shaders/introduction_to_shaders.rst:139]:
```suggestion
primitive draws. It is only available in ``spatial``, ``canvas_item``, and ``sky`` shaders.
```

**Review by clayjohn — CHANGES_REQUESTED** [review]
Looks mostly good! Just a few comments and suggestions.

File: tutorials/shaders/introduction_to_shaders.rst [hunk 4]
```diff
@@ -3,119 +3,162 @@
...
-run for all pixels in that wavefront until they are complete. Accordingly, if you have one pixel slower
-than the rest (due to excessive branching), the entire block will be slowed down, resulting in massively
-slower render times.
-
-This is different from CPU-based operations. On a CPU, if you can speed up even one
-pixel, the entire rendering time will decrease. On a GPU, you have to speed up the entire wavefront
-to speed up rendering.
+The ``fragment()`` processing function is used to set up the Godot material
+
+parameters per pixel. This code runs on every visible pixel the object or
```
Comment (NathanLovato) [inline tutorials/shaders/introduction_to_shaders.rst:138]:
```suggestion
The ``fragment()`` processing function is used to set up the Godot material
parameters per pixel. This code runs on every visible pixel the object or
```

**Comment by NathanLovato** [comment]
Thanks, just applied the changes.

**Review by clayjohn — APPROVED** [review]

**Comment by NathanLovato** [comment]
Thanks for the review @clayjohn

**Pull request merged**